it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersSoleProprietorUtrControllerISpec.scala

```diff
@@ -1,89 +1,104 @@
 package uk.gov.hmrc.fhregistrationfrontend.controllers
 
-import models.NormalMode
+import models.{CheckMode, Mode, NormalMode}
 import org.jsoup.Jsoup
+import org.scalatest.TryValues.convertTryToSuccessOrFailure
 import play.api.libs.ws.DefaultWSCookie
 import play.api.test.WsTestClient
 import play.mvc.Http.HeaderNames
+import uk.gov.hmrc.fhregistrationfrontend.forms.models.{TradingName, Utr}
+import uk.gov.hmrc.fhregistrationfrontend.pages.businessPartners.{SoleProprietorUtrPage, SoleProprietorsTradingNamePage}
 import uk.gov.hmrc.fhregistrationfrontend.testsupport.{Specifications, TestConfiguration}
 
 class BusinessPartnersSoleProprietorUtrControllerISpec
   extends Specifications with TestConfiguration {
 
-  val route: String = routes.BusinessPartnersSoleProprietorUtrController.load().url.drop(6)
+  def route(mode: Mode) = routes.BusinessPartnersSoleProprietorUtrController.load(index = 1, mode).url.drop(6)
 
-  s"GET $route" should {
+  def userAnswersWithPageData(formAnswers: Utr) = emptyUserAnswers
+    .set[Utr](SoleProprietorUtrPage(1), formAnswers)
+    .success
+    .value
 
-    "render the partnership-self-assessment-unique-taxpayer-reference page" in {
-      given
-        .commonPrecondition
+  List(NormalMode, CheckMode).foreach { mode =>
```

Comment:
Once controller is corrected you will need to test two scenarios: 

(1) Loading the page without any previous answers stored in cache 
(2) Loading the page with previous cached answers on the page